### ISSUE TITLE
About the loss during finetuning

### ISSUE THREAD

**Issue by Xiangtaokong** [FoundationVision/Infinity#54]
Thank you for your outstanding work and contributions to the open-source community. 

I am fine-tuning the model using my own data and would like to ask about how the loss should change during training. At what point should the loss indicate that the model has been trained well? 

An example training loss is: 

Lm: 0.421 (0.438) Lt: 0.436 (0.453) Accm: 79.44 (78.44) Acct: 78.85 (77.79) T: 10.469s dataT: 0.2ms. 

Is this a good result? Or should Accm be closer to 100%?

**Comment by JeyesHan** [comment]
@Xiangtaokong Thanks for your attention to Infinity. According to our experience, ~80% bit acc could yield good results.

**Comment by Xiangtaokong** [comment]
Thanks for your reply, it is very helpful. I also have another question to consult.
During training, the attention mask can lead network to focus on all previous scales.
For example:
[1, 0, 0, 0, 0, 0]
[1, 1, 1, 1, 1, 0]
[1, 1, 1, 1, 1, 0]
[1, 1, 1, 1, 1, 0]
[1, 1, 1, 1, 1, 0]
But during testing, from the code, network can only using this scale to predict the next scale res.
If we want to obtain a same operation, should the training mask also see only the current scale?
For example, using:
[1, 0, 0, 0, 0, 0]
[0, 1, 1, 1, 1, 0]
[0, 1, 1, 1, 1, 0]
[0, 1, 1, 1, 1, 0]
[0, 1, 1, 1, 1, 0]

(1 means can be seen, 0 means masked)

**Comment by JeyesHan** [comment]
@Xiangtaokong It seems that you missing the kv-cache design during inference. Actually, the training and testing attention mask is the same.

**Comment by JeyesHan** [comment]
Please refer to :
https://github.com/FoundationVision/Infinity/blob/main/infinity/models/basic.py#L292

**Comment by Xiangtaokong** [comment]
Many thanks for your reply. I did ignore kv_cache because it is updated and used inside the blocks, only being initialized at the model level. My problem is solved well.

**Comment by nuclearliu** [comment]
Thank you for open-sourcing this fantastic project!

I’m currently fine-tuning Infinity-2B, but the results so far aren’t as strong as I’d hoped. I’m curious about how low the training loss should ideally reach to achieve better performance.

At 50k steps, my training metrics are:

Lm: 0.491 (0.495)  Lt: 0.506 (0.510)  Accm: 75.39 (74.91)  Acct: 74.49 (74.00)

**Comment by WuTao-CS** [comment]
> Thank you for open-sourcing this fantastic project!
> 
> I’m currently fine-tuning Infinity-2B, but the results so far aren’t as strong as I’d hoped. I’m curious about how low the training loss should ideally reach to achieve better performance.
> 
> At 50k steps, my training metrics are:
> 
> Lm: 0.491 (0.495) Lt: 0.506 (0.510) Accm: 75.39 (74.91) Acct: 74.49 (74.00)
> 
> [@JeyesHan](https://github.com/JeyesHan) [@Xiangtaokong](https://github.com/Xiangtaokong)

Hello, I have a similar problem to yours. The  Accm and Acct stays at around 75 and cannot be improved further. Have you solved it?
